core-services/ipi-deprovision/aws.sh

```diff
@@ -44,6 +127,26 @@ if [[ -n ${HYPERSHIFT_PRUNER:-} ]]; then
 	done
 	# Exit here if we had errors, otherwise we destroy the OIDC providers for the hostedclusters and deadlock deletion as cluster api creds stop working so it will never be able to remove machine finalizers
 	if [[ $had_failure -ne 0 ]]; then exit $had_failure; fi
+	# IAM user cleanup (ci-op-* older than 72h)
+	cutoff="$(date -u -d '72 hours ago' --iso-8601=seconds)"
+	aws iam list-users --query "Users[?starts_with(UserName, 'ci-op-') && CreateDate < '${cutoff}'].UserName" --output text | tr '\t' '\n' | while read -r user; do
```

Comment:
You're currently putting this in the `HYPERSHIFT_PRUNER`.  But leaked IAM users don't seem like a HyperShift-specific thing.  Maybe you want to shift this to the main level of the script?  And you may want to move it to after the `queue deprovision` step, so that it doesn't have to get involved with the users that the installer is able to delete itself?